ui/test/nuxeo-layout.test.js

```diff
@@ -27,57 +35,130 @@ const base = url.substring(0, url.lastIndexOf('/'));
 // Export Polymer and PolymerElement for 1.x and 2.x compat
 window.Polymer = Polymer;
 
-suite.skip('<nuxeo-layout>', () => {
-  test('layout not found', async () => {
-    const layout = await fixture(
-      html`
-        <nuxeo-layout href="notfound.html"></nuxeo-layout>
-      `,
-    );
-    await waitForEvent(layout, 'iron-resize');
-    expect(layout.$$('nuxeo-error').hidden).to.be.false;
-    expect(layout.element).to.be.undefined;
-  });
+const layoutLoad = async (layout) => (!layout.element ? waitChanged(layout, 'element') : null);
 
-  test('layout stamped with model', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/layouts/dummy-layout.html" model='{"text": "dummy"}'></nuxeo-layout>
-    `);
-    await waitChanged(layout, 'element');
-    expect(layout.$$('nuxeo-error').hidden).to.be.true;
-    expect(layout.element.localName).to.equal('dummy-layout');
-    expect(layout.element.text).to.equal('dummy');
-  });
+suite('<nuxeo-layout>', () => {
+  suite('Error handling', () => {
+    test('Should display nuxeo-error when layout is not found', async () => {
+      const layout = await fixture(
+        html`
+          <nuxeo-layout href="notfound.html"></nuxeo-layout>
+        `,
+      );
+      await flush();
 
-  test('layout valid field invalid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "valid"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+      const nuxeoError = layout.$.error;
+      if (!isElementVisible(nuxeoError)) {
+        await waitForAttrMutation(nuxeoError, 'hidden', null);
+      }
+      expect(isElementVisible(layout.$.container)).to.be.false;
+      expect(isElementVisible(nuxeoError)).to.be.true;
+    });
   });
 
-  test('layout valid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "valid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(true);
-  });
+  suite('Stamping', () => {
+    test('Should map model into layout when model is defined', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model='{"text": "dummy"}'></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      expect(isElementVisible(layout.$.error)).to.be.false;
+      expect(layout.element.tagName).to.equal('DUMMY-LAYOUT');
+      expect(layout.element.shadowRoot.querySelector('span')).to.exist;
+      expect(layout.element.shadowRoot.querySelector('span').innerText).to.equal('dummy');
+      expect(layout.element.shadowRoot.querySelector('nuxeo-input')).to.exist;
+    });
+
+    test('Should stamp the layout when model is empty', async () => {
```

Comment:
Should this one be the opposite of the above one? I.e. `should not map...`?
Because stamping, in both scenarios the layout gets stamp.